### PR TITLE
Fix filtering for `id` on paginated resources

### PR DESCRIPTION
**Pull request by kmerz**
## Motivation and Context
The stream id is used all over the place when referencing a stream (e.g event message) but there is NO WAY
to find a stream based on its id.

## Description
Add a id filter option to the streams overview page and fix it for everything else.
Also add a Query Helper to stream overview.

## How Has This Been Tested?
- search with `id: 5f4dfb9c69be46153b9a9a7b`

## Screenshots (if appropriate):
Graylog - Streams (1)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Note
Suggest a backport to 3.3

### CODE REVIEW

**Review by linuspahl — COMMENTED** [review]
What do you think about mentioning the id in the query helper info for (at least) the users and teams overview as well?

File: graylog2-server/src/main/java/org/graylog2/search/SearchQueryParser.java [hunk 1]
```diff
@@ -259,6 +260,8 @@ FieldValue createFieldValue(SearchQueryField field, String quotedStringValue, bo
                 return new FieldValue(Integer.parseInt(pair.getLeft()), pair.getRight(), negate);
             case LONG:
                 return new FieldValue(Long.parseLong(pair.getLeft()), pair.getRight(), negate);
+            case ID:
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/search/SearchQueryParser.java:263]: Can you please rename this to `OBJECT_ID` to make it a bit more explicit? :slightly_smiling_face: Thank you!

**Review by bernd — CHANGES_REQUESTED** [review]
Thank you! Can you please merge in the current master branch? There are some conflicts. Thank you!

**Review by bernd — APPROVED** [review]
LGTM :+1:

**Review by linuspahl — CHANGES_REQUESTED** [review]
When a user searches for an id and there is no results, the search results in a bad request. For example: `id:search-id`. This is the case for every entity (saved searches, streams...).
It would be optimal if we keep the request result consistent, in comparision with other query fields.

**Comment by kmerz** [comment]
> When a user searches for an id and there is no results, the search results in a bad request. For example: `id:search-id`. This is the case for every entity (saved searches, streams...).
> It would be optimal if we keep the request result consistent, in comparision with other query fields.

This is because the `id:search-id` is not a valid id and the response is a illegal argument. We should handle that better but this will not part of this release anymore. I will open a separate issue for that.

**Review by linuspahl — APPROVED** [review]

**Pull request merged**